### DISCUSSION TITLE
aria2下载完成后会将文件转移到哪个目录?

### DISCUSSION THREAD

**Discussion by yhlfh** [AlistGo/alist#1775] [Q&A] [answered]
aria2和alist都是docker部署，弄了好久没搞定。我知道有一个temp dir是下载临时存放的文件，然后下载完后这个目录里就找不到东西了，alist后台aira2这里有一条“将下载的文件传输到相应的存储，显示error。请问下载完成的文件要去哪里找？

**Comment by Cub0ne** [comment]
正常来说会缓存到/opt/alist/data/temp/aria2/下面

**Reply by yhlfh** [reply]
还在下载中的文件是在这里面没错。下载完后就不见了。

**Reply by Cub0ne** [reply]
可以用AriaNg看一下下载目录是哪个，我是用docker启动的，是在这里
![image](https://user-images.githubusercontent.com/30786071/191937559-01fd48d1-b3c7-4ae7-ac92-7e9ba9d2750a.png)

**Reply by orzonzofz** [reply]
你的2个容器没做映射，没有互通。你的东西下载到aria2的容器里了，它上传也找不到alist里的目标目录，肯定报错了。

**Reply by Cub0ne** [reply]
已经做了映射，我的没问题
![image](https://user-images.githubusercontent.com/30786071/191953242-8993c38c-92ca-4692-b802-e17ebee67a98.png)

**Reply by orzonzofz** [reply]
抱歉，我回答的是楼主

**Reply by yhlfh** [reply]
我映射了。能正常下载。下载中的临时文件也位于temp目录。下载完成后temp目录里就没东西了。而且我注意到，容器重新启动会删除temp目录里的东西（我试过把宿主机文件夹映射到temp，结果那个文件夹里的东西全没了）。所以我猜想temp只是暂存的目录，下载完成后文件会被转移到别的目录。感谢楼上二位的回答，但是你们没答到点上。我现在不是不能能过aria2下载，而是下载完毕以后找不着文件了。

**Reply by yhlfh** [reply]
> 你的2个容器没做映射，没有互通。你的东西下载到aria2的容器里了，它上传也找不到alist里的目标目录，肯定报错了。

alist里的目标目录是哪个？我看到alist默认只有/opt/alist/data目录是做映射的。

**Reply by yhlfh** [reply]
> 已经做了映射，我的没问题
> ![image](https://user-images.githubusercontent.com/30786071/191953242-8993c38c-92ca-4692-b802-e17ebee67a98.png)

如图所示界面的下方应该还有一个项目叫“将下载的文件传输到相应的存储”，那里显示的是什么？

**Reply by Cub0ne** [reply]
之前的删除掉了，刚才测试了一下，可以上传，aria2的下载目录和alist的保存目录映射一致就行了，比如我的都是 `-v /opt/software/aria2/downloads:/opt/alist/data/temp`
![image](https://user-images.githubusercontent.com/30786071/191980828-cec2b460-ece9-40fb-b3a9-34dcd44b1ba5.png)

**Reply by yhlfh** [reply]
> 之前的删除掉了，刚才测试了一下，可以上传，aria2的下载目录和alist的保存目录映射一致就行了，比如我的都是 `-v /opt/software/aria2/downloads:/opt/alist/data/temp`
> ![image](https://user-images.githubusercontent.com/30786071/191980828-cec2b460-ece9-40fb-b3a9-34dcd44b1ba5.png)

下载完成后的文件还是在/opt/alist/data/temp这目录？你重启下alist2的container看看文件还在吗？

**Reply by Cub0ne** [reply]
传输完会自动删除的

**Reply by yhlfh** [reply]
> 传输完会自动删除的

啥意思？下载完成后自动删除文件？

**Reply by Cub0ne** [reply]
> > 传输完会自动删除的
> 
> 啥意思？下载完成后自动删除文件？

上传完成之后会自动删除

**Reply by yhlfh** [reply]
> > > 传输完会自动删除的
> > 
> > 
> > 啥意思？下载完成后自动删除文件？
> 
> 上传完成之后会自动删除

哦。我的情况是下载。下载完找不着文件。

**Reply by Cub0ne** [reply]
> > > > 传输完会自动删除的
> > > 
> > > 
> > > 啥意思？下载完成后自动删除文件？
> > 
> > 
> > 上传完成之后会自动删除
> 
> 哦。我的情况是下载。下载完找不着文件。

下载的文件在Aria2容器里面，`/opt/alist/data/temp`路径下

**Reply by orzonzofz** [reply]
大哥，是我没说到点子上，还是你没看明白，你的2个容器内部文件互通吗？
你要把aria2的下载目录映射到宿主机，再把宿主机的这个目录映射到你的alist里，2个容器都不互通，aria2肯定能正常下载，但是它找不到alist容器里的目录它往哪里传？要么报错，要么它在aria2容器里建了alist的目录，传到那里去了，并删了你下载东西，还看不懂没办了。

**Reply by yhlfh** [reply]
> 大哥，是我没说到点子上，还是你没看明白，你的2个容器内部文件互通吗？
> 你要把aria2的下载目录映射到宿主机，再把宿主机的这个目录映射到你的alist里，2个容器都不互通，aria2肯定能正常下载，但是它找不到alist容器里的目录它往哪里传？要么报错，要么它在aria2容器里建了alist的目录，传到那里去了，并删了你下载东西，还看不懂没办了。

互通的。就是如你所说宿主机到两个容器都映射了。下载的东西两个容器都能访问到，下载未完成时临时文件都有的。下载完成没有任何报错，但是，完成的文件找不着了。

**Reply by loryncien** [reply]
> > 大哥，是我没说到点子上，还是你没看明白，你的2个容器内部文件互通吗？
> > 你要把aria2的下载目录映射到宿主机，再把宿主机的这个目录映射到你的alist里，2个容器都不互通，aria2肯定能正常下载，但是它找不到alist容器里的目录它往哪里传？要么报错，要么它在aria2容器里建了alist的目录，传到那里去了，并删了你下载东西，还看不懂没办了。
> 
> 互通的。就是如你所说宿主机到两个容器都映射了。下载的东西两个容器都能访问到，下载未完成时临时文件都有的。下载完成没有任何报错，但是，完成的文件找不着了。

aria2映射好了alist的下载临时文件，也能下载完成，**完成后移动到了aria2容器的`downloads`**，在其对应的宿主机映射目录可看见，这实际上相当于完成了远程离线下载。
同时要想将下载的文件在alist进行文件管理，则再将宿主机的下载目录映射到alist并添加本地存储挂载

**Reply by yhlfh** [reply]
看了你的回复，我终于明白AList界面右下角展开菜单中的“添加到Aria2 下载”是什么意思了。我在哪个目录执行这个操作，文件就会被放到那个目录。原来是如此操作的啊！我之前测试下载的时候，都是在测试文件所在目录执行这个操作，然后下完文件就被放回了原位，难怪一直找不着下载完成的文件。也就是说，假设我要把百度网盘上的某文件下载到本机，要添加一个本机存储，然后在AList本机存储目录下执行“添加到Aria2 下载”这个操作。

**Reply by loryncien** [reply]
> 看了你的回复，我终于明白AList界面右下角展开菜单中的“添加到Aria2 下载”是什么意思了。我在哪个目录执行这个操作，文件就会被放到那个目录。原来是如此操作的啊！我之前测试下载的时候，都是在测试文件所在目录执行这个操作，然后下完文件就被放回了原位，难怪一直找不着下载完成的文件。也就是说，假设我要把百度网盘上的某文件下载到本机，要添加一个本机存储，然后在AList本机存储目录下执行“添加到Aria2 下载”这个操作。

Alist的Aria2有两个功能：
一个是本地下载(本地设置里配置)，选择文件，`发送到Aria2`，即通过本地Aria2下载，如Motrix；
一个是远程下载(在管理后台配置)，对应功能`添加到Aria2 下载`，即下载文件到部署Alist的（远程）主机上，正确挂载后方便管理；

**Reply by yhlfh** [reply]
嗯。就是不知道这个远程下载的应用场景是什么。下载到aria2所在的机器上，然后又上传到云，想不出什么情况下要这么干。

**Comment by yhlfh** [comment, marked as answer]
https://github.com/alist-org/alist/discussions/1785
这里作者本人答复了。我的问题已解决，感谢各位热心网友！

**Reply by 793457961** [reply]
怎么解决的

**Comment by sjtuross** [comment]
感谢楼上各位的讨论，终于明白了2个不同的利用Aria2的工作流程。简单总结，一个是模拟网盘离线下载的功能（这里所谓网盘也可以是某个本地挂载目录），一个是推送到Aria2下载。因为都用到了Aria2，而且都能做到下载到本地，真是我中有你你中有我。
